aggregative_results/aggregators/energy_model_aggregator.py

```
@@ -0,0 +1,155 @@
+            process_features=process_features,
+            system_features=system_features
+        )
+
+    def process_sample(self, sample: EnergyModelFeatures) -> Union[EnergyModelResult, EmptyAggregationResults]:
+        try:
+            if self.__previous_sample is None:
+                return EmptyAggregationResults()
+
+            sample_as_dict = self.__convert_sample_to_dict(sample)
+            duration = sample.timestamp - self.__previous_sample.timestamp
+            sample_as_dict[DURATION_COLUMN] = duration
+            sample_as_df = pd.DataFrame([sample_as_dict])
+            energy_prediction = self.__model.predict(sample_as_df)
+
+            energy_per_resource = self.__calculate_energy_per_resource(sample, energy_prediction)
+            return EnergyModelResult(energy_mwh=energy_prediction,
+                                     cpu_energy_consumption=energy_per_resource.cpu_energy_consumption,
+                                     ram_energy_consumption=energy_per_resource.ram_energy_consumption,
+                                     disk_io_read_energy_consumption=energy_per_resource.disk_io_read_energy_consumption,
+                                     disk_io_write_energy_consumption=energy_per_resource.disk_io_write_energy_consumption,
+                                     network_io_received_energy_consumption=energy_per_resource.network_io_received_energy_consumption,
+                                     network_io_sent_energy_consumption=energy_per_resource.network_io_sent_energy_consumption)
+        except Exception as e:
+            return EmptyAggregationResults()
+        finally:
+            self.__previous_sample = sample
+
+    def __convert_sample_to_dict(self, sample: EnergyModelFeatures) -> dict[str, any]:
+        sample_dict = {**asdict(sample.process_features), **asdict(sample.system_features)}
+        return {key: value for key, value in sample_dict.items() if value is not None}
```

Are you sure?
You are going to send absent columns to the model to predict that way.

---

not really. All values should not be None. Maybe i should raise an error